whois-api/src/main/java/net/ripe/db/whois/api/rdap/RdapObjectMapper.java

```diff
@@ -156,6 +159,63 @@ public Object mapSearch(final String requestUrl, final List<RpslObject> objects,
         return mapCommons(searchResult, requestUrl);
     }
 
+    public Object mapMergeSearch(final String requestUrl, List<RpslObject> objects,
+                                 final Iterable<LocalDateTime> localDateTimes, final int maxResultSize){
+        RdapObject organisation = new RdapObject();
+        List<Ip> ipv4Networks = new ArrayList<>();
+        List<Ip> ipv6Networks = new ArrayList<>();
+        List<Autnum> autnums = new ArrayList<>();
+        final Iterator<LocalDateTime> iterator = localDateTimes.iterator();
+        Notice outOfLimitNotice = null;
+
+        if(objects.size() > maxResultSize){
+            List<RpslObject> autnumRpsl = objects.stream().filter(rpsl -> ObjectType.AUT_NUM.equals(rpsl.getType()))
+                    .collect(Collectors.toList());
+            List<RpslObject> organisationRpsl =
+                    objects.stream().filter(rpsl -> ObjectType.ORGANISATION.equals(rpsl.getType()))
+                    .collect(Collectors.toList());
+            List<RpslObject> ipv4Rpsl = TopLevelFilter.filter(objects.stream().filter(rpsl -> ObjectType.INETNUM.equals(rpsl.getType()))
+                    .collect(Collectors.toList()), ObjectType.INETNUM);
+            List<RpslObject> ipv6Rpsl = TopLevelFilter.filter(objects.stream().filter(rpsl -> ObjectType.INET6NUM.equals(rpsl.getType()))
+                    .collect(Collectors.toList()), ObjectType.INET6NUM);
+
+            objects = new ArrayList<>();
+            objects.addAll(autnumRpsl);
+            objects.addAll(organisationRpsl);
+            objects.addAll(ipv4Rpsl);
+            objects.addAll(ipv6Rpsl);
+
+            outOfLimitNotice = new Notice();
+            outOfLimitNotice.setTitle(String.format("limited search results to %s maximum" , maxResultSize));
+            organisation.getNotices().add(outOfLimitNotice);
+        }
+
+
+        for (final RpslObject object : objects) {
+            if (object.getType() == AUT_NUM ) {
+                autnums.add((Autnum) getRdapObject(requestUrl, object, iterator.next(),
+                        Optional.empty()));
+            } else if (object.getType() == ObjectType.INETNUM) {
+                ipv4Networks.add((Ip) getRdapObject(requestUrl, object, iterator.next(),
+                        Optional.empty()));
+            } else if (object.getType() == INET6NUM ) {
+                ipv6Networks.add((Ip) getRdapObject(requestUrl, object, iterator.next(),
+                        Optional.empty()));
+            } else if (object.getType() == ObjectType.ORGANISATION) {
+                organisation = getRdapObject(requestUrl, object, iterator.next(), Optional.empty());
+            }
+        }
+
+        organisation.setAutnums(autnums);
+        organisation.setNetworks(ipv4Networks, ipv6Networks);
+
+        if(outOfLimitNotice != null){
+            organisation.getNotices().add(outOfLimitNotice);
```

Comment:
`outOfLimitNotice` is added *twice*. I suggest removing the `Notice outOfLimitNotice = null` above and the not null code here.